src/settings.rs

```diff
@@ -146,43 +100,6 @@ impl sequential_storage::map::Key for SettingsKey {
     }
 }
 
-#[derive(
-    Default, serde::Serialize, serde::Deserialize, Clone, PartialEq, Eq,
-)]
-pub struct SettingsItem(Vec<u8, 256>);
-
-impl<'a> sequential_storage::map::Value<'a> for SettingsItem {
-    fn serialize_into(
-        &self,
-        buffer: &mut [u8],
-    ) -> Result<usize, SerializationError> {
-        if let Some(buf) = buffer.get_mut(..self.0.len()) {
-            buf.copy_from_slice(&self.0);
-            Ok(self.0.len())
-        } else {
-            Err(SerializationError::BufferTooSmall)
-        }
-    }
-
-    fn deserialize_from(buffer: &'a [u8]) -> Result<Self, SerializationError> {
-        Vec::from_slice(buffer)
-            .map_err(|_| SerializationError::BufferTooSmall)
-            .map(Self)
-    }
-}
-
-#[derive(Debug)]
-pub enum Error<F> {
-    Postcard(postcard::Error),
-    Flash(F),
-}
-
-impl<F> From<postcard::Error> for Error<F> {
-    fn from(e: postcard::Error) -> Self {
-        Self::Postcard(e)
-    }
-}
-
 pub struct SerialSettingsPlatform<C, const Y: usize> {
```

Comment:
I think the only thing that differs in the implementation here between Stabilizer and Booster is the `platform` command - is there some way we could move most of this code into `serial-settings` for ease of use? Can do this in a follow-up PR, but I noticed there's a lot of duplication between the two projects.

Reply:
The serial port, the flash, the impl details of the kv storage on top of the flash, and the platform cmd impls differ. That's very much what platform is.

Reply:
https://github.com/quartiq/stabilizer/issues/909 for the rest